### PR TITLE
Enable FutureWarnings/DeprecationWarnings as errors for dask_cudf

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Part of https://github.com/rapidsai/build-planning/issues/26

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Review by galipremsagar — APPROVED** [review]

**Review by bdice — APPROVED** [review]

**Comment by rjzamora** [comment]
Sorry - Just noticed that this PR may clash with https://github.com/rapidsai/cudf/pull/15593. Pretty sure my last commit will avoid the problem, but I will keep my eye on CI and follow up if I missed anything.

**Comment by mroeschke** [comment]
@rjzamora let me know if you'd like me to wait on https://github.com/rapidsai/cudf/pull/15593 before merging this PR

(Note I had to adjust a `FutureWarning` in cudf to better match pandas and prevent a false-positive warning for dask-cudf)

File: python/dask_cudf/dask_cudf/io/tests/test_json.py [hunk 1]
```diff
@@ -84,9 +84,9 @@ def test_read_json_nested(tmp_path):
         }
     )
     kwargs = dict(orient="records", lines=True)
-    with tmp_path / "data.json" as f, dask.config.set(
-        {"dataframe.convert-string": False}
-    ):
+    f = tmp_path / "data.json"
+    f.touch()
```
Comment (wence-) [inline python/dask_cudf/dask_cudf/io/tests/test_json.py:88]: *question*: Why do we need to touch the file first?

File: python/dask_cudf/dask_cudf/tests/test_accessor.py [hunk 2]
```diff
@@ -543,7 +543,7 @@ def test_struct_explode(data):
 
 
 def test_tz_localize():
-    data = Series(date_range("2000-04-01", "2000-04-03", freq="H"))
+    data = Series(date_range("2000-04-01", "2000-04-03", freq="h"))
```
Comment (wence-) [inline python/dask_cudf/dask_cudf/tests/test_accessor.py:546]: Were these "always wrong"?

**Review by wence- — APPROVED** [review]

File: python/dask_cudf/dask_cudf/io/tests/test_json.py (hunk 1 above)
Comment (mroeschke) [reply python/dask_cudf/dask_cudf/io/tests/test_json.py]: Ah turns out we didn't need to touch the file. Removed

File: python/dask_cudf/dask_cudf/tests/test_accessor.py (hunk 2 above)
Comment (mroeschke) [reply python/dask_cudf/dask_cudf/tests/test_accessor.py:546]: Not "always wrong" I think these aliases are just deprecated in pandas 2.x

**Pull request merged**